pkg/addon/addon.go

```diff
@@ -1438,3 +1445,51 @@ func fetchVelaCoreImageTag(ctx context.Context, k8sClient client.Client) (string
 	}
 	return tag, nil
 }
+
+// PackageAddon package vela addon directory into a helm chart compatible archive and return its absolute path
+func PackageAddon(addonDictPath string) (string, error) {
+	meta := &Meta{}
+	metaData, err := ioutil.ReadFile(filepath.Clean(filepath.Join(addonDictPath, MetadataFileName)))
+	if err != nil {
+		return "", err
+	}
+
+	err = yaml.Unmarshal(metaData, meta)
+	if err != nil {
+		return "", err
+	}
+
+	chartFile := &chart.Metadata{
+		Name:        meta.Name,
+		Description: meta.Description,
+		// define Vela addon's type to be library in order to prevent installation of a common chart. Please refer to https://helm.sh/docs/topics/library_charts/
+		Type:       "library",
+		Version:    meta.Version,
+		AppVersion: meta.Version,
+		APIVersion: chart.APIVersionV2,
+		Icon:       meta.Icon,
+		Home:       meta.URL,
+		Keywords:   meta.Tags,
```

Comment:
what's the convert relationship between `metatdata.yaml` and `chart.yaml`? Please make sure we didn't miss any useful information here. \cc @wangyikewxgm

Reply:
we need to add a doc into `kubevela.io` to describe these details for guiding addon developers.

Reply:
> what's the convert relationship between `metatdata.yaml` and `chart.yaml`? Please make sure we didn't miss any useful information here. \cc @wangyikewxgm

Have updated description of this PR to explain the mapping. @wangyikewxgm pls review it.

Reply:
> we need to add a doc into `kubevela.io` to describe these details for guiding addon developers.

OK. I will update document later.

Reply:
LGTM, the translate rule is totally same with the logic in catalog. https://github.com/kubevela/catalog/blob/4b46deb768b887e809934049f5aad6cefaa32a8b/hack/addons/syn_addon_package.go#L108

Reply:
Please make this info clear in the doc. This file is only be used for generate the index.yaml for chart-museum automatically in "package command". This file is useless when enable an addon. So the addon's maker  needn't take aware of the exist of this file and mustn't write this file.

Reply:
the CI in catalog should use the same code with this PR after merged @wangyikewxgm

Reply:
> the CI in catalog should use the same code with this PR after merged @wangyikewxgm

Yes, It's true.

Reply:
> 



> Please make this info clear in the doc. This file is only be used for generating the index.yaml for chart-museum automatically .It's useless when enable an addon. So the addon's maker needn't take aware of the existence of this file and mustn't write this file.

OK. Thanks for reminder.